server/src/main/java/org/elasticsearch/health/cluster/CoordinationHealthIndicatorService.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+package org.elasticsearch.health.cluster;
+
+import org.elasticsearch.cluster.ClusterChangedEvent;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.ClusterStateApplier;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.cluster.node.DiscoveryNodeRole;
+import org.elasticsearch.health.HealthIndicator;
+import org.elasticsearch.health.HealthIndicatorService;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class CoordinationHealthIndicatorService implements HealthIndicatorService, ClusterStateApplier {
+
+    private AtomicReference<List<HealthIndicator>> indicators = new AtomicReference<>(List.of());
+
+    @Override
+    public List<HealthIndicator> getIndicators() {
+        return indicators.get();
+    }
+
+    @Override
+    public void applyClusterState(ClusterChangedEvent event) {
+        if (event.localNodeMaster() == false) {
+            indicators = new AtomicReference<>(List.of());
+            return;
+        }
+        indicators = new AtomicReference<>(List.of(noEligibleMasterNodes(event.state())));
+    }
+
+    private HealthIndicator noEligibleMasterNodes(ClusterState clusterState) {
+        for (DiscoveryNode node : clusterState.nodes()) {
+            if (node.getRoles().contains(DiscoveryNodeRole.MASTER_ROLE)) {
+                return new NoEligibleMasterNodes(HealthIndicator.Status.GREEN, "Eligible master nodes");
+            }
+        }
+        return new NoEligibleMasterNodes(HealthIndicator.Status.RED, "No eligible master nodes");
```

Comment:
Likewise I think I missed a discussion about the design of this indicator. Does it ever return `RED` after a node has joined a cluster? What is the user supposed to do about it?

Reply:
I think this comment was marked as resolved in error since it relates to code being added in this PR.